### PR TITLE
Adds the Cleric's Den ruin, and Datum Maces.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request
This PR adds Maces as a new weapon, obtainable from a new space ruin, the Cleric's Den. The Cleric's den is a fairly simple ruin, part of a larger station that crumbled apart due to the reckless usage of supposedly holy magic. ...Or something like that. Contains a flask of holy water, as well as an Iron mace, and a design disc allowing for the creation of a cleric's mace in a similar way to the knight's armor, in all the fun datum material flavors. 
ClericsDen

I'm not particularly attached to the damage values/material costs on the maces, so if it needs to be toned down I'm more than happy to hear suggestions. In particular, the material costs SEEM fine for everything EXCEPT uranium, which can create a fairly ridiculous amount of radiation in a few swings.

MaceShowcase
Showcase of a few of the different maces in different materials. (Updated 10/7/19)

Additionally, the ruin is guarded by a new weaker cult construct, the Proteon. It's not meant for cult gameplay, just serving as a basic hostile mob, which are primarily spawned by triggering the traps on the ruin. Might be interesting to toy around with after the antag freeze, but for now they're exceptionally basic.

## Why It's Good For The Game
Thought it would make for good companion content relative to the Wizard's Tower Lavaland Ruin. Also, I find it odd that with all the plate armor, chainmail, spears, swords, clubs, daggers, staves, and every other medieval weapon we have on the codebase, we still didn't have any maces. There are dozens of us cleric players, DOZENS.

## Changelog
:cl:
add: An old monastery from a previously abandoned sector of space has recently resurfaced in some regions surrounding the station.
add: Latent scans of the surrounding systems have picked up trace signs of new, smaller cult constructs, however these signatures quickly vanished.
add: In other news, scavengers in your sector have been seen occasionally with classically recreated maces, so autolathe firmware has been upgraded to accommodate this.
/:cl:

### CODE REVIEW

**Comment by PKPenguin321** [comment]
Could use better sprites since that's definitely a stun baton but it seems cool nonetheless

**Comment by nemvar** [comment]
Hooo boy, time to implement smithing.

**Comment by Tlaltecuhtli** [comment]
simple smithing:
- new mineral clay: you make casts with it in hand
- blacksmith oven: put ores get molted mineral (its a reagent)
- pour molten mineral in cast
- splash water in cast to get thing
- flavour things like putting the thing back in oven then on an anvil/hit with hammer/ water for rng +1 dmg

**Comment by nemvar** [comment]
Or like just implement something basic that people can actually use without looking at the wiki for 20 minutes.

**Comment by Tlaltecuhtli** [comment]
souless autolathe crafting? 😪

**Comment by Ty-the-Smonk** [comment]
More medieval shit with datums would be neat

**Comment by MMMiracles** [comment]
>datum smithing

oh no its the synthflesh shovels all over again

**Comment by Fikou** [comment]
maybe one of those crusader zombies from snowdin should be here? i feel like there should be some enemy

**Comment by ArcaneMusic** [comment]
> maybe one of those crusader zombies from snowdin should be here? i feel like there should be some enemy

Ohoho, don't worry, I got some magic in the works.

**Comment by Shadowflame909** [comment]
@MMMiracles  you haven't suffered until you've been hit by my skewium sword

**Comment by Potato-Masher** [comment]
Lifeweb blacksmith, now coming to TG.

**Comment by lolman360** [comment]
excuse me its clearly 
image

**Comment by ExcessiveUseOfCobblestone** [comment]
bet you still make bronze bars noob

**Comment by ArcaneMusic** [comment]
> bet you still make bronze bars noob


Imagine not using blast furnace mithryl training in the year of our lord 2k19 🤔

File: code/modules/ruins/spaceruin_code/clericsden.dm [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+/obj/item/paper/fluff/ruins/clericsden/contact
+	info = "Father Aurellion, the ritual is complete, and soon our brothers at the bastion will see the error of our ways. After all, a god of clockwork or blood? Preposterous. Only the TRUE GOD should have so much power. Signed, Father Odivallus."
+
+/obj/item/paper/fluff/ruins/clericsden/warning
+	info = "FATHER ODIVALLUS DO NOT GO FORWARD WITH THE RITUAL. THE ASTEROID WE'RE ANCHORED TO IS UNSTABLE, YOU WILL DESTROY THE STATION. I HOPE THIS REACHES YOU IN TIME. FATHER AURELLION."
+
+/mob/living/simple_animal/hostile/construct/proteon
+	name = "Proteon"
+	real_name = "Proteon"
+	desc = "A weaker construct meant to scour ruins for objects of Nar'Sie's affection. Those barbed claws are no joke."
```
Comment (ATH1909) [inline code/modules/ruins/spaceruin_code/clericsden.dm]: wait, doesn't one of the fluff messages mention that this monastery ISN'T dedicated to nar-sie? why are Her constructs here?

Comment (ArcaneMusic) [reply code/modules/ruins/spaceruin_code/clericsden.dm]: It's implied that whatever holy ritual they performed backfired, summoning or attracting constructs to the monastery, and destroying the attached station.

File: code/game/objects/items/melee/misc.dm [hunk 2]
```diff
@@ -633,3 +633,29 @@
...
+	throwforce = 8
+	block_chance = 10
+	armour_penetration = 50
+	attack_verb = list("smacked", "struck", "cracked", "beaten")
+	var/overlay_state = "mace_handle"
+	var/mutable_appearance/overlay
+
+/obj/item/melee/cleric_mace/Initialize()
+	. = ..()
+	overlay = mutable_appearance(icon, overlay_state, ABOVE_OBJ_LAYER)
```
Comment (nemvar) [inline code/game/objects/items/melee/misc.dm]: Remove the layer. You want this one float layer

Comment (ArcaneMusic) [reply code/game/objects/items/melee/misc.dm]: Gotcha.

**Comment by ArcaneMusic** [comment]
Alright, as of now I was able to get the handle of the mace sprite to color independantly from the datum material color. Right now it's working with overlays, and I have been told and reminded that it should probably be using blends, but I haven't had any luck with that so far.
image
That being said, this is likely ready now.

**Comment by ArcaneMusic** [comment]
I'm only going to have time to resolve the merge conflict this weekend, It'll be a hot second before I can touch this.

**Comment by ArcaneMusic** [comment]
Cool, this is ready for review again. (Hopefully before someone decides to change more than 25,000 lines of weapon code by removing a game mode)

**Pull request merged**